code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -191,3 +198,386 @@
+	randomdir = FALSE
+	duration = 9
+	pixel_z = 270
+
+/obj/effect/temp_visual/gonarch_artillery_projectile/Initialize(mapload)
+	. = ..()
+	animate(src, pixel_z = 0, time = duration)
+
+/obj/effect/temp_visual/gonarch_artillery/proc/fall(list/flame_hit)
+	var/turf/T = get_turf(src)
+	playsound(T, 'sound/magic/fleshtostone.ogg', 80, TRUE)
+	new /obj/effect/temp_visual/gonarch_artillery_projectile(T)
+	sleep(duration)
+	playsound(T, "sound/effects/splat.ogg", 80, TRUE)
+	var/datum/reagents/R = new/datum/reagents(20)
+	R.my_atom = src
+	R.add_reagent("sacid" , 20)
+	chem_splash(T, splash_range, list(R), 0, 1, FALSE)
+	for(var/mob/living/C in range(splash_range, src))
+		to_chat(C, "<span class='userdanger'>You have been splashed with acid!</span>")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/mode_switch(mode_to_switch, sound_to_play, clear_targets = TRUE, change_icon)
+	debug_gonarch_modeswitches.Add(mode_to_switch) //TODO: Remove simple debug
+	if(mode_to_switch == mode)
+		return
+	mode = mode_to_switch
+	if(clear_targets)
+		LoseTarget()
+	if(sound_to_play)
+		var/sound_picked = pick(sound_to_play)
+		playsound(src, sound_picked, 200, TRUE, 15, pressure_affected = FALSE)
+	if(change_icon)
+		icon_state = change_icon
+		icon_living = change_icon
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rest()
+	if(get_area(src) == home_nest_area) // If I am home & Everything is calm
+		if(aggro_vision_range != gonarch_aggro_vision_range || vision_range != gonarch_standard_vision_range || speed != gonarch_standard_speed)
+			aggro_vision_range = gonarch_aggro_vision_range
+			vision_range = gonarch_standard_vision_range
+			speed = gonarch_standard_speed
+		if(homesick_level)
+			homesick_level = 0
+			step_towards(src, home_nest_turf, speed)
+		make_headcrabs() //Its its own proc, so they can be called by admins.
+	else
+		if(homesick_level >= 5)
+			mode_switch(GONARCH_MODE_RETURNING, gonarch_soundlist_return, change_icon = "gonarch_peaceful")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_return()
+	speed = -1
+	var/turf/olddist = get_dist(src, home_nest_turf)
+	step_towards(src, home_nest_turf, speed)
+	target = home_nest_turf
+	if(get_dist(src, target) >= olddist)
+		frustration++
+		if(frustration > frustration_limit)
+			DestroyPathToTarget()
+			homesick_level++
+	else
+		frustration = 0
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_newhome()
+	var/list/turfs = list()
+	var/list/turfs_in_range = range(4, src)
+	for(var/turf/T in range(gonarch_nestfind_range, src))
+		if(T in turfs_in_range)
+			continue
+		if(isspaceturf(T))
+			continue
+		if(T.density)
+			continue
+		var/lightingcount = T.get_lumcount(0.5) * 10
+		if(lightingcount > 1 || gonarch_finding_home_attempts >= gonarch_finding_home_limit_attempts)
+			continue
+		turfs += T
+	if(length(turfs) <= 0)
+		gonarch_nestfind_range += 5
+		gonarch_finding_home_attempts++
+		return
+	home_nest_turf = pick(turfs)
```

Comment:
```suggestion
	gonarch_finding_home_attempts = 0
	home_nest_turf = pick(turfs)
```
You'd most likely want to reset this once a turf is actually found. Else it will always consider illuminated tiles as its new home when they once had issues with finding a home.